### ISSUE TITLE
Cannot use default system-trust with deb-packaged scylla

### ISSUE THREAD

**Issue by xemul** [scylladb/scylladb#13904]
S3 client uses tls::creds_builder::set_system_trust() to configure http client's socket. However, when scylla is build using `dist-server-deb` on Fedora host and then the package is installed on AWS, the TLS socket connects without root certificates.

Steps-to-reproduce

0. Patch scylla with a patch below
1. Build scylla .deb package with `ninja dist-server-deb`
2. Create instance on AWS from scylla-5.3 AMI
3. Upload built .deb and install it
4. Run scylla from installed deb (need object-storage.conf)

Result:

```
[shard  0] init - Startup failed: seastar::tls::verification_error (The certificate is NOT trusted. The certificate issuer is unknown. (Issuer=[C=US,O=Amazon,CN=Amazon RSA 2048 M01], Subject=[CN=*.s3.us-east-2.amazonaws.com]))
```

The patch

```patch
--- a/main.cc
+++ b/main.cc
@@ -100,6 +100,7 @@
 #include "service/raft/raft_group_registry.hh"
 #include "service/raft/raft_group0_client.hh"
 #include "service/raft/raft_group0.hh"
+#include "utils/s3/client.hh"
 
 #include <boost/algorithm/string/join.hpp>
 
@@ -704,6 +705,16 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
 
             startlog.info(startup_msg, scylla_version(), get_build_id());
 
+            {
+                startlog.info("Getting S3 endpoint config");
+                s3::endpoint_config epcfg = cfg->object_storage_config().at("s3.us-east-2.amazonaws.com");
+                startlog.info("Making S3 client");
+                auto cln = s3::client::make("s3.us-east-2.amazonaws.com", make_lw_shared<s3::endpoint_config>(std::move(epcfg)));
+                startlog.info("Requesting object size");
+                auto sz = cln->get_object_size("/xemul/shards").get0();
+                startlog.info("-> {}", sz);
+            }
+
             // Set the default scheduling_group, i.e., the main scheduling
             // group to a lower shares. Subsystems needs higher shares
             // should set it explicitly. This prevents code that is supposed to
```

**Comment by xemul** [comment]
Strace output:

```
5156  write(2, "INFO  2023-05-16 11:44:12,144 [shard 0] init - Scylla version 5.3.0~dev-0.20230516.1ebc8021b202 with build-id 6765c3d41c5457155ec448a2ed14e106a18296d6 starting ...\n\n", 165) = 165
5156  write(2, "INFO  2023-05-16 11:44:12,144 [shard 0] init - Getting S3 endpoint config\n", 80) = 80
5156  write(2, "INFO  2023-05-16 11:44:12,144 [shard 0] init - Making S3 client\n", 70) = 70
5156  openat(AT_FDCWD, "/etc/resolv.conf", O_RDONLY) = 7
5156  newfstatat(7, "", {st_mode=S_IFREG|0644, st_size=931, ...}, AT_EMPTY_PATH) = 0
5156  read(7, "# This is /run/systemd/resolve/stub-resolv.conf managed by man:systemd-resolved(8).\n# Do not edit.\n#\n# This file might be symlinked as /etc/resolv.conf. If you're looking at\n# /etc/resolv.conf and seeing this text, you have followed the symlink.\n#\n# This i"..., 4096) = 931
5156  read(7, "", 4096)                 = 0
5156  close(7)                          = 0
5156  openat(AT_FDCWD, "/dev/urandom", O_RDONLY) = 7
5156  read(7, "\314\276\202\227\333Qig,\346\r\251O\320j\204\177\327\v\n\261\203OFf9%\310\245\317\351", 31) = 31
5156  close(7)                          = 0
5156  openat(AT_FDCWD, "/etc/hosts", O_RDONLY) = 7
5156  newfstatat(7, "", {st_mode=S_IFREG|0644, st_size=221, ...}, AT_EMPTY_PATH) = 0
5156  read(7, "127.0.0.1 localhost\n\n# The following lines are desirable for IPv6 capable hosts\n::1 ip6-localhost ip6-loopback\nfe00::0 ip6-localnet\nff00::0 ip6-mcastprefix\nff02::1 ip6-allnodes\nff02::2 ip6-allrouters\nff02::3 ip6-allhosts\n", 4096) = 221
5156  read(7, "", 4096)                 = 0
5156  close(7)                          = 0
5156  socket(AF_INET, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, IPPROTO_IP) = 7
5156  setsockopt(7, SOL_IP, IP_PKTINFO, "\1", 1) = 0
5156  bind(7, {sa_family=AF_INET, sin_port=htons(0), sin_addr=inet_addr("0.0.0.0")}, 128) = 0
5156  getsockname(7, {sa_family=AF_INET, sin_port=htons(54688), sin_addr=inet_addr("0.0.0.0")}, [128 => 16]) = 0
5156  write(2, "INFO  2023-05-16 11:44:12,146 [shard 0] init - Requesting object size\n", 76) = 76
5156  write(2, "TRACE 2023-05-16 11:44:12,146 [shard 0] s3 - HEAD /xemul/shards\n", 64) = 64
5156  write(2, "DEBUG 2023-05-16 11:44:12,146 [shard 0] s3 - Waiting for factory to initialize\n", 79) = 79
5156  read(6, "\1\0\0\0\0\0\0\0", 8)    = 8
5156  timerfd_settime(6, TFD_TIMER_ABSTIME, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=2596, tv_nsec=335990033}}, NULL) = 0
5156  io_submit(0x7f9dffffa000, 3, [{aio_data=0x6000001d5450, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=7, aio_buf=POLLOUT}, {aio_data=0x6000001d53e8, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=7, aio_buf=POLLIN}, {aio_data=0x6000002e0200, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=6, aio_buf=POLLIN}]) = 3
5156  read(4, "D\1\0\0\0\0\0\0", 8)     = 8
5156  read(6, 0x7ffcdcff7a88, 8)        = -1 EAGAIN (Resource temporarily unavailable)
5156  io_submit(0x7f9e00308000, 2, [{aio_data=0x6000002e0180, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=6, aio_buf=POLLIN}, {aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 2
5156  sendmsg(7, {msg_name={sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("127.0.0.53")}, msg_namelen=16, msg_iov=[{iov_base="\310j\1\0\0\1\0\0\0\0\0\0\2s3\tus-east-2\tamazonaws\3com\0\0\1\0\1", iov_len=44}], msg_iovlen=1, msg_controllen=0, msg_flags=0}, 0) = 44
5156  timerfd_settime(4, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=0}}, NULL) = 0
5156  timer_settime(0, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=0}}, NULL) = 0
5156  membarrier(MEMBARRIER_CMD_PRIVATE_EXPEDITED, 0) = 0
5156  timerfd_settime(6, TFD_TIMER_ABSTIME, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=2592, tv_nsec=148171303}}, NULL) = 0
5156  rt_sigprocmask(SIG_SETMASK, ~[RTMIN RT_1], ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], 8) = 0
5156  read(4, 0x7ffcdcff7aa8, 8)        = -1 EAGAIN (Resource temporarily unavailable)
5156  io_submit(0x7f9dffffa000, 1, [{aio_data=0x6000002e0260, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=3, aio_buf=POLLIN}]) = 1
5156  rt_sigprocmask(SIG_SETMASK, ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], NULL, 8) = 0
5156  timerfd_settime(4, 0, {it_interval={tv_sec=0, tv_nsec=500000}, it_value={tv_sec=0, tv_nsec=500000}}, NULL) = 0
5156  io_submit(0x7f9e00308000, 1, [{aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 1
5156  timer_settime(0, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=1300, tv_nsec=11920}}, NULL) = 0
5156  recvmsg(7, {msg_name={sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("127.0.0.53")}, msg_namelen=128 => 16, msg_iov=[{iov_base="\310j\201\200\0\1\0\1\0\0\0\0\2s3\tus-east-2\tamazonaws\3com\0\0\1\0\1\300\f\0\1\0\1\0\0\0\2\0\0044\333m9", iov_len=65507}], msg_iovlen=1, msg_control=[{cmsg_len=20, cmsg_level=SOL_IP, cmsg_type=IP_PKTINFO}], msg_controllen=20, msg_flags=MSG_CTRUNC}, 0) = 60
5156  socket(AF_INET, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, IPPROTO_IP) = 8
5156  setsockopt(8, SOL_IP, IP_PKTINFO, "\1", 1) = 0
5156  bind(8, {sa_family=AF_INET, sin_port=htons(0), sin_addr=inet_addr("0.0.0.0")}, 128) = 0
5156  getsockname(8, {sa_family=AF_INET, sin_port=htons(60445), sin_addr=inet_addr("0.0.0.0")}, [128 => 16]) = 0
5156  getsockname(2, 0x60000027590c, [16]) = -1 ENOTSOCK (Socket operation on non-socket)
5156  shutdown(8, SHUT_RD)              = -1 ENOTCONN (Transport endpoint is not connected)
5156  shutdown(8, SHUT_WR)              = -1 ENOTCONN (Transport endpoint is not connected)
5156  close(8)                          = 0
5156  shutdown(7, SHUT_RD)              = -1 ENOTCONN (Transport endpoint is not connected)
5156  shutdown(7, SHUT_WR)              = -1 ENOTCONN (Transport endpoint is not connected)
5156  close(7)                          = 0
5156  read(4, "\2\0\0\0\0\0\0\0", 8)    = 8
5156  io_submit(0x7f9e00308000, 1, [{aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 1
5156  write(2, "DEBUG 2023-05-16 11:44:12,151 [shard 0] s3 - Initialized factory, address=52.219.109.57:443 tls=yes\n", 100) = 100
5156  write(2, "DEBUG 2023-05-16 11:44:12,151 [shard 0] s3 - Making new HTTPS connection addr=52.219.109.57:443 host=s3.us-east-2.amazonaws.com\n", 128) = 128
5156  socket(AF_INET, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, IPPROTO_TCP) = 7
5156  setsockopt(7, SOL_SOCKET, SO_REUSEADDR, [0], 4) = 0
5156  setsockopt(7, SOL_IP, IP_BIND_ADDRESS_NO_PORT, [1], 4) = 0
5156  bind(7, {sa_family=AF_INET, sin_port=htons(65477), sin_addr=inet_addr("0.0.0.0")}, 16) = 0
5156  connect(7, {sa_family=AF_INET, sin_port=htons(443), sin_addr=inet_addr("52.219.109.57")}, 16) = -1 EINPROGRESS (Operation now in progress)
5156  io_submit(0x7f9dffffa000, 1, [{aio_data=0x6000001d5450, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=7, aio_buf=POLLOUT}]) = 1
5156  timerfd_settime(4, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=0}}, NULL) = 0
5156  timer_settime(0, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=0}}, NULL) = 0
5156  membarrier(MEMBARRIER_CMD_PRIVATE_EXPEDITED, 0) = 0
5156  timerfd_settime(6, TFD_TIMER_ABSTIME, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=2592, tv_nsec=148108405}}, NULL) = 0
5156  rt_sigprocmask(SIG_SETMASK, ~[RTMIN RT_1], ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], 8) = 0
5156  read(4, 0x7ffcdcff7aa8, 8)        = -1 EAGAIN (Resource temporarily unavailable)
5156  rt_sigprocmask(SIG_SETMASK, ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], NULL, 8) = 0
5156  timerfd_settime(4, 0, {it_interval={tv_sec=0, tv_nsec=500000}, it_value={tv_sec=0, tv_nsec=500000}}, NULL) = 0
5156  timerfd_settime(4, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=0}}, NULL) = 0
5156  timer_settime(0, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=0}}, NULL) = 0
5156  membarrier(MEMBARRIER_CMD_PRIVATE_EXPEDITED, 0) = 0
5156  rt_sigprocmask(SIG_SETMASK, ~[RTMIN RT_1], ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], 8) = 0
5156  io_pgetevents(0x7f9dffffa000, 1, 128, [{data=0x6000001d5450, obj=0x6000001d5410, res=4, res2=0}], NULL, {sigmask=~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], sigsetsize=8}) = 1
5156  rt_sigprocmask(SIG_SETMASK, ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], NULL, 8) = 0
5156  timerfd_settime(4, 0, {it_interval={tv_sec=0, tv_nsec=500000}, it_value={tv_sec=0, tv_nsec=500000}}, NULL) = 0
5156  io_submit(0x7f9e00308000, 1, [{aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 1
5156  timer_settime(0, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=1300, tv_nsec=11920}}, NULL) = 0
5156  getsockopt(7, SOL_SOCKET, SO_ERROR, [0], [4]) = 0
5156  newfstatat(AT_FDCWD, "/opt/scylladb/libreloc/gnutls.config", {st_mode=S_IFREG|0755, st_size=3076, ...}, 0) = 0
5156  getsockname(7, {sa_family=AF_INET, sin_port=htons(65477), sin_addr=inet_addr("10.99.17.90")}, [128 => 16]) = 0
5156  write(2, "TRACE 2023-05-16 11:44:12,163 [shard 0] http - created new http connection 10.99.17.90:65477\n", 93) = 93
5156  io_submit(0x7f9dffffa000, 1, [{aio_data=0x6000001d54b8, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=7, aio_buf=POLLRDHUP}]) = 1
5156  read(4, "\1\0\0\0\0\0\0\0", 8)    = 8
5156  io_submit(0x7f9e00308000, 1, [{aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 1
5156  rt_sigprocmask(SIG_BLOCK, NULL, ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], 8) = 0
5156  rt_sigprocmask(SIG_SETMASK, ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], NULL, 8) = 0
5156  openat(AT_FDCWD, "/etc/pkcs11/pkcs11.conf", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
5156  getuid()                          = 1000
5156  openat(AT_FDCWD, "/home/scyllaadm/.config/pkcs11/pkcs11.conf", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
5156  openat(AT_FDCWD, "/home/scyllaadm/.config/pkcs11/modules", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
5156  openat(AT_FDCWD, "/etc/pkcs11/modules", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
5156  openat(AT_FDCWD, "/usr/share/p11-kit/modules", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
5156  getrandom("\xfa\x97\x13\x94\x02\x48\xd7\x7c\x4c\xf7\x2f\x85\x42\xb5\xcf\x9d\x92\xff\xde\x8f\x5b\xad\xfa\x1b\x09\xe3\x95\xb2\xd9\x76\x4f\xa1\x58\xcc\x40\xe5\x52\x23\x49\x32\xb6\x1c\x31\x4f\xd1\x52\x34\xb1\x25\xe1\xe1\xf4\x38\x99\x7c\xb2\x20\xda\xa0\xf9\xfd\xa4\xe4\xfd", 64, 0) = 64
5156  io_submit(0x7f9dffffa000, 1, [{aio_data=0x6000001d5450, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=7, aio_buf=POLLOUT}]) = 1
5156  read(4, "\3\0\0\0\0\0\0\0", 8)    = 8
5156  io_submit(0x7f9e00308000, 1, [{aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 1
5156  sendmsg(7, {msg_name=NULL, msg_namelen=0, msg_iov=[{iov_base="\26\3\3\1\213\1\0\1\207\3\3T}\34*\317\335)\341g\373\242\374\307'w1G\200\252*\7\217:t|_0Ia\f\304\325 z0\211a4<b)\363\202\31\23\227\216\243&2\233\274B?\0312\215Zn\217\207\33\350\201A\0:\23\2\23\3\23\1\23\4\3000\314\250\300\24\300/\300\23\300,\300\255\314\251\300\n\300+\300\254\300\t\0\235\300\235\0005\0\234\300\234\0/\0\237\300\237\314\252\09\0\236\300\236\0003\1\0\1\4\0\5\0\5\1\0\0\0\0\0\n\0\26\0\24\0\35\0\36\0\27\0\30\0\31\1\0\1\1\1\2\1\3\1\4\0\v\0\2\1\0\0\r\0\36\0\34\4\3\5\3\6\3\10\7\10\10\10\t\10\n\10\v\10\4\10\5\10\6\4\1\5\1\6\1\0\26\0\0\0\27\0\0\0#\0\0\0003\0k\0i\0\35\0 )\365\225\216\230\233)8\372\241+\331\227\256\262L\2668\331"..., iov_len=400}], msg_iovlen=1, msg_controllen=0, msg_flags=0}, MSG_NOSIGNAL) = 400
5156  io_submit(0x7f9dffffa000, 1, [{aio_data=0x6000001d53e8, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=7, aio_buf=POLLIN}]) = 1
5156  timerfd_settime(4, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=0}}, NULL) = 0
5156  timer_settime(0, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=0}}, NULL) = 0
5156  membarrier(MEMBARRIER_CMD_PRIVATE_EXPEDITED, 0) = 0
5156  rt_sigprocmask(SIG_SETMASK, ~[RTMIN RT_1], ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], 8) = 0
5156  read(4, 0x7ffcdcff7aa8, 8)        = -1 EAGAIN (Resource temporarily unavailable)
5156  rt_sigprocmask(SIG_SETMASK, ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], NULL, 8) = 0
5156  timerfd_settime(4, 0, {it_interval={tv_sec=0, tv_nsec=500000}, it_value={tv_sec=0, tv_nsec=500000}}, NULL) = 0
5156  timerfd_settime(4, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=0}}, NULL) = 0
5156  timer_settime(0, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=0}}, NULL) = 0
5156  membarrier(MEMBARRIER_CMD_PRIVATE_EXPEDITED, 0) = 0
5156  rt_sigprocmask(SIG_SETMASK, ~[RTMIN RT_1], ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], 8) = 0
5156  io_pgetevents(0x7f9dffffa000, 1, 128, [{data=0x6000001d53e8, obj=0x6000001d53a8, res=195, res2=0}], NULL, {sigmask=~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], sigsetsize=8}) = 1
5156  rt_sigprocmask(SIG_SETMASK, ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], NULL, 8) = 0
5156  timerfd_settime(4, 0, {it_interval={tv_sec=0, tv_nsec=500000}, it_value={tv_sec=0, tv_nsec=500000}}, NULL) = 0
5156  io_submit(0x7f9e00308000, 1, [{aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 1
5156  timer_settime(0, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=1300, tv_nsec=11920}}, NULL) = 0
5156  recvfrom(7, "\26\3\3\0_\2\0\0[\3\3\317\357\179Z\363\342\241(\354d\344\305\265\177\321\277\222l\217\375\t8-.\310\240\200\257\16\365\202 \317\216u\332y\267k\257\2375yr\33\243\221\242\275\230\232\247\33G\v\312\212.\3\177i\3044\206\300/\0\0\23\0\0\0\0\0\v\0\2\1\0\377\1\0\1\0\0\27\0\0\26\3\3\25\200\v\0\25|\0\25y\0\7\3740\202\7\3700\202\6\340\240\3\2\1\2\2\20\1_\377\n\205\306\323l1\37\244\363D\215\265\3310\r\6\t*\206H\206\367\r\1\1\v\5\0000<1\v0\t\6\3U\4\6\23\2US1\0170\r\6\3U\4\n\23\6Amazon1\0340\32\6\3U\4\3\23\23Amazon RSA 2048 M010\36\27\r230411000000Z\27\r240228235959Z0"..., 8192, MSG_DONTWAIT, NULL, NULL) = 5609
5156  openat(AT_FDCWD, "/etc/gnutls/pkcs11.conf", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
5156  read(4, "\4\0\0\0\0\0\0\0", 8)    = 8
5156  io_submit(0x7f9e00308000, 1, [{aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 1
5156  read(4, "\1\0\0\0\0\0\0\0", 8)    = 8
5156  io_submit(0x7f9e00308000, 1, [{aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 1
5156  read(4, "\3\0\0\0\0\0\0\0", 8)    = 8
5156  io_submit(0x7f9e00308000, 1, [{aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 1
5156  getsockname(7, {sa_family=AF_INET, sin_port=htons(65477), sin_addr=inet_addr("10.99.17.90")}, [128 => 16]) = 0
5156  write(2, "TRACE 2023-05-16 11:44:12,181 [shard 0] http - dropping connection 10.99.17.90:65477\n", 85) = 85
5156  shutdown(7, SHUT_RD)              = 0
5156  shutdown(7, SHUT_WR)              = 0
5156  getsockname(7, {sa_family=AF_INET, sin_port=htons(65477), sin_addr=inet_addr("10.99.17.90")}, [128 => 16]) = 0
5156  write(2, "WARN  2023-05-16 11:44:12,181 [shard 0] seastar - Exceptional future ignored: seastar::tls::verification_error (The certificate is NOT trusted. The certificate issuer is unknown. (Issuer=[C=US,O=Amazon,CN=Amazon RSA 2048 M01], Subject=[CN=*.s3.us-east-2.am"..., 7573) = 7573
5156  read(4, "\2\0\0\0\0\0\0\0", 8)    = 8
5156  io_submit(0x7f9e00308000, 1, [{aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 1
5156  write(2, "TRACE 2023-05-16 11:44:12,182 [shard 0] http - destroyed connection 10.99.17.90:65477\n", 86) = 86
5156  close(7)                          = 0
5156  write(2, "INFO  2023-05-16 11:44:12,182 [shard 0] init - Shutting down sighup\n", 68) = 68
5156  rt_sigprocmask(SIG_BLOCK, NULL, ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], 8) = 0
5156  rt_sigprocmask(SIG_SETMASK, ~[HUP INT QUIT ILL ABRT FPE KILL SEGV ALRM TERM CONT STOP TSTP TTIN TTOU RTMIN RT_1 RT_3], NULL, 8) = 0
5156  read(4, "\2\0\0\0\0\0\0\0", 8)    = 8
5156  io_submit(0x7f9e00308000, 1, [{aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 1
5156  write(2, "INFO  2023-05-16 11:44:12,183 [shard 0] init - Shutting down sighup was successful\n", 83) = 83
5156  write(2, "INFO  2023-05-16 11:44:12,183 [shard 0] init - Shutting down configurables\n", 75) = 75
5156  read(4, "\1\0\0\0\0\0\0\0", 8)    = 8
5156  io_submit(0x7f9e00308000, 1, [{aio_data=0x6000002e0128, aio_lio_opcode=IOCB_CMD_POLL, aio_fildes=4, aio_buf=POLLIN}]) = 1
5156  write(2, "INFO  2023-05-16 11:44:12,183 [shard 0] init - Shutting down configurables was successful\n", 90) = 90
5156  write(2, "ERROR 2023-05-16 11:44:12,183 [shard 0] init - Startup failed: seastar::tls::verification_error (The certificate is NOT trusted. The certificate issuer is unknown. (Issuer=[C=US,O=Amazon,CN=Amazon RSA 2048 M01], Subject=[CN=*.s3.us-east-2.amazonaws.com]))\n", 256) = 256
5156  exit_group(1)                     = ?
5157  <... read resumed> <unfinished ...>) = ?
5157  +++ exited with 1 +++
5156  +++ exited with 1 +++
```

It tries to access neither `CENTOS_SSL_CERT_FILE="/etc/pki/tls/cert.pem"` nor `DEBIAN_SSL_CERT_FILE="/etc/ssl/certs/ca-certificates.crt"`

**Comment by xemul** [comment]
@elcallio , need your help here

**Comment by avikivity** [comment]
Tip: use -e trace=file to show only syscalls that reference filenames.

**Comment by avikivity** [comment]
https://www.gnutls.org/manual/gnutls.html#gnutls_005fcertificate_005fset_005fx509_005fsystem_005ftrust

But, it will run into the problem of using the wrong files on debian.

**Comment by xemul** [comment]
> So we need to tell it to use the system trust ourselves.

We already do:

```c++
            [state, use_https] () -> future<> {
                if (use_https) {
                    tls::credentials_builder cbuild;
                    co_await cbuild.set_system_trust();
                    state->creds = cbuild.build_certificate_credentials();
                }
            }
```

**Comment by xemul** [comment]
@avikivity , the problem is _not_ that scylla tries wrong file for root certificates, the problem is that scylla doesn't try any file at all for some reason. Here's the `-e trace=file` output.

```
$ cat strace.output | fgrep -v '"/sys' | fgrep -v '"/proc' | fgrep -v '""' | fgrep -v '"/opt/scylladb'
execve("/usr/bin/scylla", ["scylla", "-c1", "--logger-log-level", "s3=trace:http=trace"], 0x7ffed6042cc0 /* 22 vars */) = 0
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libtinfo.so.6", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/dev/tty", O_RDWR|O_NONBLOCK) = 3
openat(AT_FDCWD, "/usr/lib/locale/locale-archive", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/share/locale/locale.alias", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_IDENTIFICATION", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_IDENTIFICATION", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/usr/lib/x86_64-linux-gnu/gconv/gconv-modules.cache", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_MEASUREMENT", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_MEASUREMENT", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_TELEPHONE", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_TELEPHONE", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_ADDRESS", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_ADDRESS", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_NAME", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_NAME", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_PAPER", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_PAPER", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_MESSAGES", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_MESSAGES", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_MESSAGES/SYS_LC_MESSAGES", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_MONETARY", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_MONETARY", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_COLLATE", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_COLLATE", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_TIME", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_TIME", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_NUMERIC", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_NUMERIC", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_CTYPE", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/locale/C.utf8/LC_CTYPE", O_RDONLY|O_CLOEXEC) = 3
newfstatat(AT_FDCWD, "/home/scyllaadm", {st_mode=S_IFDIR|0750, st_size=4096, ...}, 0) = 0
newfstatat(AT_FDCWD, ".", {st_mode=S_IFDIR|0750, st_size=4096, ...}, 0) = 0
newfstatat(AT_FDCWD, "/home", {st_mode=S_IFDIR|0755, st_size=4096, ...}, 0) = 0
newfstatat(AT_FDCWD, "/home/scyllaadm", {st_mode=S_IFDIR|0750, st_size=4096, ...}, 0) = 0
openat(AT_FDCWD, "/usr/bin/scylla", O_RDONLY) = 3
newfstatat(AT_FDCWD, "/usr/bin/scylla", {st_mode=S_IFREG|0755, st_size=444, ...}, 0) = 0
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
access("/etc/system-fips", F_OK)        = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/scyllaadm/.config/seastar/seastar.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/scyllaadm/.config/seastar/io.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib64/hwloc", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/localtime", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/etc/resolv.conf", O_RDONLY) = 7
openat(AT_FDCWD, "/dev/urandom", O_RDONLY) = 7
openat(AT_FDCWD, "/etc/hosts", O_RDONLY) = 7
openat(AT_FDCWD, "/etc/pkcs11/pkcs11.conf", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/scyllaadm/.config/pkcs11/pkcs11.conf", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/scyllaadm/.config/pkcs11/modules", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/pkcs11/modules", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/share/p11-kit/modules", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/gnutls/pkcs11.conf", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
+++ exited with 1 +++
```

 When the same binary is run on "native" host OS, it also fails to find pkcs11 files, but still goes and just opens the /etc/pki/tls/cert.pem one

**Comment by xemul** [comment]
Another option is to set trust file by its debian path explicitly as was done in the 1st version of #13493

**Comment by mykaul** [comment]
cc @fruch

**Comment by fruch** [comment]
> A-ha. Solved with
> 
> ```
> # apt install p11-kit p11-kit-modules
> # mkdir /usr/lib64/pkcs11
> # ln -s /usr/lib/x86_64-linux-gnu/pkcs11/p11-kit-trust.so /usr/lib64/pkcs11/p11-kit-trust.so
> ```

I've solved it by just deleting p11-kit-trust.so from the relocatable folder, but then we need to make sure those packages are installed, and hope there isn't a breaking change in newer versions.

Anyhow this block testing of EaR KMS

**Comment by kreuzerkrieg** [comment]
So, all we need is something like

```
Subject: [PATCH] client: Refactor `make_request` calls

Change `make_request` in such a way that it will be not aware of the request type it is making, specifically make it unaware of MPU completion call
---
Index: dist/debian/control.template
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/dist/debian/control.template b/dist/debian/control.template
--- a/dist/debian/control.template	(revision cb0e4a7358277276950192cc10600d7f49a5a127)
+++ b/dist/debian/control.template	(date 1729589867770)
@@ -17,7 +17,7 @@
 
 Package: %{product}-server
 Architecture: any
-Depends: ${misc:Depends}, %{product}-conf (= ${binary:Version}), %{product}-python3 (= ${binary:Version})
+Depends: ${misc:Depends}, %{product}-conf (= ${binary:Version}), %{product}-python3 (= ${binary:Version}), p11-kit, p11-kit-modules
 Replaces: %{product}-tools (<<5.5)
 Breaks: %{product}-tools (<<5.5)
 Description: Scylla database server binaries 
Index: dist/debian/debian/rules
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/dist/debian/debian/rules b/dist/debian/debian/rules
--- a/dist/debian/debian/rules	(revision cb0e4a7358277276950192cc10600d7f49a5a127)
+++ b/dist/debian/debian/rules	(date 1729592421117)
@@ -17,6 +17,9 @@
 
 override_dh_auto_clean:
 
+override_dh_install:
+	dh_install -Xlibp11-kit
+
 override_dh_auto_install:
 	dh_auto_install
 	./scylla/install.sh --packaging --root "$(CURDIR)/debian/tmp" $(install_arg) --sysconfdir "/etc/default"

```

@syuu1228 WDYT?

**Comment by tchaikov** [comment]
discussed with Ernest offline. he had concerns on the missing dependencies in scylla's AMI image after adding `p11-kit` and `p11-kit-modules` to debian/control. but after checking 

- https://github.com/scylladb/scylla-machine-image/blob/next/packer/build_image.sh
- https://github.com/scylladb/scylla-machine-image/blob/next/aws/cloudformation/scylla.yaml.j2

i cannot find the places where we track the dependencies out side of debian/control. i think we install scylla packages *and* their dependencies every time we build an AMI images instead of preparing their dependencies separately *before* installing scylla packages. further more, we can verify our changes with our existing workflows: we do build:

- [AMI image](https://jenkins.scylladb.com/job/scylla-master/job/ami/)
- [GCE image](https://jenkins.scylladb.com/job/scylla-master/job/gce-image/)
- [Azure image](https://jenkins.scylladb.com/job/scylla-master/job/azure-image/)

in [unified-deb](https://jenkins.scylladb.com/job/scylla-master/job/unified-deb/) pipeline, which is in turn triggered by the [daily job](https://jenkins.scylladb.com/job/scylla-master/job/daily/).

**Comment by tchaikov** [comment]
and, i was reading https://github.com/scylladb/scylla-enterprise/issues/2792. the gist is:

- https://github.com/p11-glue/p11-kit/issues/502 does not help. despite that the change has been included by fedora 40. the changes source file compiles into a sample. what matters is

  * compile-time: `--module-path` when building `p11-kit`. or 
  * runtime: `"module"` setting in `pkcs11.conf`
- we override p11-kit config path in ent release as a fix of https://github.com/scylladb/scylla-enterprise/issues/2792. but the fix is not included by oss.

IMO, the right way to go is to have the fix overriding p11-kit config path.

**Comment by regevran** [comment]
@tchaikov, if by 
> IMO, the right way to go is to have the fix overriding p11-kit config path.

You mean cherry-picking [55909](https://github.com/scylladb/scylla-enterprise/commit/55909ac45e91eaf4e2a9393d0738f4febbc28739#diff-3be036231354cce9942706688e5d72d98333c64624ee75477e038107ca42f2ef) then it seems OK to me.

**Comment by fruch** [comment]
> So, all we need is something like
> 
> ```
> diff --git a/dist/debian/control.template b/dist/debian/control.template
> index 5deb3b7638..6e37aed8be 100644
> --- a/dist/debian/control.template
> +++ b/dist/debian/control.template
> @@ -17,7 +17,7 @@ Conflicts: %{product}-server (<< 1.1)
>  
>  Package: %{product}-server
>  Architecture: any
> -Depends: ${misc:Depends}, %{product}-conf (= ${binary:Version}), %{product}-python3 (= ${binary:Version})
> +Depends: ${misc:Depends}, %{product}-conf (= ${binary:Version}), %{product}-python3 (= ${binary:Version}), p11-kit, p11-kit-modules
>  Replaces: %{product}-tools (<<5.5)
>  Breaks: %{product}-tools (<<5.5)
>  Description: Scylla database server binaries 
> diff --git a/dist/debian/debian/rules b/dist/debian/debian/rules
> index 20a81b38f2..121e602be1 100755
> --- a/dist/debian/debian/rules
> +++ b/dist/debian/debian/rules
> @@ -17,6 +17,9 @@ override_dh_auto_build:
>  
>  override_dh_auto_clean:
>  
> +override_dh_install:
> +	dh_install -Xlibp11-kit
> +
>  override_dh_auto_install:
>  	dh_auto_install
>  	./scylla/install.sh --packaging --root "$(CURDIR)/debian/tmp" $(install_arg) --sysconfdir "/etc/default"
> ```
> 
> [@syuu1228](https://github.com/syuu1228) WDYT?

that won't work in offline installer, and not in dtests which are based on that
plus it doesn't cover the RPM installation.

we've been in this whole story for KMS support, I agree with @tchaikov, that we should just take all what was done into OSS
I think it should have went into the OSS to begin with, but now we have lots of the reason to port it back.

**Comment by syuu1228** [comment]
> So, all we need is something like
> 
> ```
> diff --git a/dist/debian/control.template b/dist/debian/control.template
> index 5deb3b7638..6e37aed8be 100644
> --- a/dist/debian/control.template
> +++ b/dist/debian/control.template
> @@ -17,7 +17,7 @@ Conflicts: %{product}-server (<< 1.1)
>  
>  Package: %{product}-server
>  Architecture: any
> -Depends: ${misc:Depends}, %{product}-conf (= ${binary:Version}), %{product}-python3 (= ${binary:Version})
> +Depends: ${misc:Depends}, %{product}-conf (= ${binary:Version}), %{product}-python3 (= ${binary:Version}), p11-kit, p11-kit-modules
>  Replaces: %{product}-tools (<<5.5)
>  Breaks: %{product}-tools (<<5.5)
>  Description: Scylla database server binaries 
> diff --git a/dist/debian/debian/rules b/dist/debian/debian/rules
> index 20a81b38f2..121e602be1 100755
> --- a/dist/debian/debian/rules
> +++ b/dist/debian/debian/rules
> @@ -17,6 +17,9 @@ override_dh_auto_build:
>  
>  override_dh_auto_clean:
>  
> +override_dh_install:
> +	dh_install -Xlibp11-kit
> +
>  override_dh_auto_install:
>  	dh_auto_install
>  	./scylla/install.sh --packaging --root "$(CURDIR)/debian/tmp" $(install_arg) --sysconfdir "/etc/default"
> ```
> 
> [@syuu1228](https://github.com/syuu1228) WDYT?

No, it's complecated since we use relocatable package.
We don't need p11-kit in debian/control dependencies, we actually provide p11-kit from relocatable pacakge.
We don't use `dh_install -Xlib* `, we implement everything in install.sh for dh_install stage.
What acutally need to do is https://github.com/scylladb/scylladb/commit/55909ac45e91eaf4e2a9393d0738f4febbc28739.

**Comment by tchaikov** [comment]
> Maybe I need to backport [55909ac](https://github.com/scylladb/scylladb/commit/55909ac45e91eaf4e2a9393d0738f4febbc28739) to OSS version?

Yeah, we can backport that commit, but I recommend waiting unless we have a compelling or urgent reason to apply this fix.

**Comment by regevran** [comment]
@tchaikov - can you please cherry-pick (backport) this commit?

**Comment by regevran** [comment]
> [@tchaikov](https://github.com/tchaikov) - can you please cherry-pick (backport) this commit?
> 
> To my understanding this is what holds us from having a clean installation (without a development environment) that can use Native Scylla backup and restore.

But maybe it is [already done](https://github.com/scylladb/scylladb/pull/22233/commits/a3af2c68e0558aa10380acdf53c300df655c775a)?

**Comment by tchaikov** [comment]
it's not. i will help with the backport once @elcallio 's patchset lands.
